### DISCUSSION TITLE
Service Example

### DISCUSSION THREAD

**Discussion by seikosantana** [scottoffen/grapevine#94] [Q&A] [answered]
Hi, i noticed context.Services and i wonder how is the use case and example of services as it sounds powerful.

Is there any example or guide i can follow?
Thanks in advance

**Comment by scottoffen** [comment, marked as answer]
It's for dependency injection. You shouldn't need to touch it in your routes, but the router uses it to resolve dependencies when running the routes (such as injecting loggers). You can see it in action in the Sample project.

- In [`Startup.cs`](https://github.com/scottoffen/grapevine/blob/main/src/Samples/Startup.cs#L23) I configure services to be injected, such as logging and HttpClient.

- In my resources, you can see that I [inject a logger](https://github.com/scottoffen/grapevine/blob/main/src/Samples/Resources/FormDataResource.cs#L13), even if I don't use it.

- In the [GitHub resource](https://github.com/scottoffen/grapevine/blob/main/src/Samples/Clients/GitHubClient.cs#L14), I use the configured HttpClient to act as a pass-though to the GitHub api.

All this, via the magic of dependency injection!
